pid_cache.gemspec

```diff
@@ -19,6 +19,8 @@ Gem::Specification.new do |spec|
   spec.metadata["source_code_uri"] = spec.homepage
   spec.metadata["changelog_uri"] = File.join(spec.homepage, "blob/master/CHANGELOG.md")
 
+  spec.required_ruby_version = ">= 3.1.0"
```

Comment:
This makes no sense. The whole point of the gem is to give a perf boot to older Ruby version, if you bump the required version you make it useless.

Reply:
The gem rely on an API that is only present on 3.1+, on older rubies it's a noop. 

But requiring Ruby 3.1 to add it is counter productive, because the goal of making it a noop, is that you can include it in your gemfile or gemspec regardless. There is no reason to prevent it on an older ruby if it's not gonna break anything.

Reply:
That makes sense, thank you for explaining. My initial reason for adding `required_ruby_version` was to ensure that the `.rubocop.yml` configuration could automatically infer the Ruby version from this specification. However, we can omit it and allow RuboCop to autonomously resolve the target Ruby version using its fallback mechanisms.